recipes/yasm/all/conanfile.py

```diff
@@ -14,87 +19,83 @@ class YASMConan(ConanFile):
     license = "BSD-2-Clause"
     settings = "os", "arch", "compiler", "build_type"
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
+    def export_sources(self):
+        export_conandata_patches(self)
+
     def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
-    def build_requirements(self):
-        if self._settings_build.os == "Windows" and self.settings.compiler != "Visual Studio" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+    def layout(self):
+        if is_msvc(self):
+            cmake_layout(self)
+        else:
+            basic_layout(self)
 
     def package_id(self):
         del self.info.settings.compiler
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version][0],
-                  destination=self._source_subfolder, strip_root=True)
-        tools.download(**self.conan_data["sources"][self.version][1],
-                       filename=os.path.join(self._source_subfolder, "YASM-VERSION-GEN.bat"))
+        get(self, **self.conan_data["sources"][self.version][0],
+                  destination=self.source_folder, strip_root=True)
 
     @property
     def _msvc_subfolder(self):
```

Comment:
This property can be removed

Reply:
done here: b4167c4c967058976885e8a0b4b4c2711c7823e1